Add optional facet values and counts search to listing endpoint

njohner opened the pull request:
For the new GEVER UI we need the possibility to optionally query for facet available values on the listing endpoint. 

resolves #5738 

## Checkliste

- [x] Changelog-Eintrag vorhanden/nötig?
- [x] Aktualisierung Dokumentation vorhanden/nötig?

- [x] Still need to bump ftw.solr instead of installing from source once it is released

deiferni commented on opengever/api/listing.py:
```diff
@@ -340,24 +344,40 @@ def solr_results(self, name, term, columns, start, rows, sort_on,
-        )
+
+        # We map the index names back to the field names for the facets
+        facet_counts = {}
```
i would add one more nesting level to this dict as with these changes we are now providing meta-information in the response, too. if i read correctly at the moment we would return name as key and count as value in the toplevel dict. i would prefer if facet counts are were all available under their own dict-key, e.g. `resp['facets']['fieldname_1']`. That way we can add more field meta-information to the response in the future without conflicts.

lukasgraf commented:
Necessary changes on the `ftw.solr` side have been released with `ftw.solr = 2.5.0`

njohner commented:
Ok so I added a comment to show where the nesting is done in the response.

Moreover, @all4manu asked me to also return the labels for the facets, as they do not want to display the tokens, but the labels. This avoids them having to send a second request to get the vocabularies or such. I've added this in a separate commit. Unfortunately I don't think we can write a test for that as long as we cannot properly test solr.

njohner commented:
Example response:
```
{
    "@id": "http://localhost:8080/fd/ordnungssystem/fuehrung/@listing?name=documents&facets%3Alist=review_state&facets%3Alist=sequence_number",
    "b_size": 25,
    "b_start": 0,
    "facets": {
        "review_state": {
            "document-state-draft": {
                "count": 3,
                "label": "nicht definiert"
            }
        },
        "sequence_number": {
            "2": {
                "count": 1
            },
            "33": {
                "count": 1
            },
            "35": {
                "count": 1
            }
        }
    },
    "items": [
        {
            "@id": "http://localhost:8080/fd/ordnungssystem/fuehrung/kommunikation/allgemeines/dossier-1/document-1"
        },
        {
            "@id": "http://localhost:8080/fd/ordnungssystem/fuehrung/gemeinderecht/dossier-14/document-35"
        },
        {
            "@id": "http://localhost:8080/fd/ordnungssystem/fuehrung/gemeinderecht/dossier-14/document-33"
        }
    ],
    "items_total": 3
}
```
